### PR TITLE
Changed abstraction implementations

### PR DESCRIPTION
**Pull request by uncleempty**
To be merged after https://github.com/oat-sa/tao-core/pull/2571

### CODE REVIEW

File: src/Watcher/ElasticDocumentBuilderFactory.php [hunk 1]
```diff
@@ -21,24 +21,24 @@
...
 use oat\tao\elasticsearch\Watcher\Resources\TesttakerIndexDocumentBuilder;
 use oat\tao\elasticsearch\Watcher\Resources\UnclassifiedIndexDocumentBuilder;
 use oat\tao\model\search\index\DocumentBuilder\DocumentBuilderFactoryInterface;
-use oat\tao\model\search\index\DocumentBuilder\GenerisIndexDocumentBuilder;
 use oat\tao\model\search\index\DocumentBuilder\IndexDocumentBuilderInterface;
 use oat\tao\model\TaoOntology;
 
 class ElasticDocumentBuilderFactory implements DocumentBuilderFactoryInterface
 {
     public const AVAILABLE_DOCUMENT_BUILDERS = [
```
Comment (andreluizmachado) [inline src/Watcher/ElasticDocumentBuilderFactory.php:34]: Please, consider removing this constant.

File: src/Watcher/Resources/GenericIndexDocumentBuilder.php [hunk 2]
```diff
@@ -26,36 +26,4 @@
 
 class GenericIndexDocumentBuilder extends AbstractIndexDocumentBuilder
```
Comment (andreluizmachado) [inline src/Watcher/Resources/GenericIndexDocumentBuilder.php:27]: Do we still need this class?

Comment (uncleempty) [reply src/Watcher/Resources/GenericIndexDocumentBuilder.php:27]: Same answer as for main PR

Comment (andreluizmachado) [reply src/Watcher/Resources/GenericIndexDocumentBuilder.php:27]: Hi, I wrote an answer there :)

**Pull request merged**